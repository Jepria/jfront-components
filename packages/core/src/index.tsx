export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
