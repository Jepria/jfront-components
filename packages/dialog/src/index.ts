export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'