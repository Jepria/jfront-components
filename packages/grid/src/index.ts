export * from './common'
export * from './jepgrid'
