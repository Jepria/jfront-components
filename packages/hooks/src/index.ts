export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
