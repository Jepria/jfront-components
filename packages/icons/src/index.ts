export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
