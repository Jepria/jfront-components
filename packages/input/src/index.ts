export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
