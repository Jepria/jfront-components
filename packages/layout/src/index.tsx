import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
