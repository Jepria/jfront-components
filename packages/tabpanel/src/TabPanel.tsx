import styled from "styled-components"
import React from "react"

const TabPanel = styled.div`
  height: 22px;
  font-family: tahoma, arial, helvetica, sans-serif;
  color: rgb(21, 66, 139);
  font-size: 11px;
  z-index: 1;
`

interface TabProps {
  selected?: boolean
}

const Tab = styled.div<TabProps>`
  display: inline-block;
  height: 16px;
  text-align: center;
  margin-left: 2px;
  padding: 3px 6px 3px 6px;
  min-width: 20px;
  border: 1px solid #8db2e3;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  border-bottom-color: #d7e4f3;
  ${(props) =>
    props.selected
      ? `
  background-color: #D7E4F3;
    cursor: default;
    font-weight: bold;
  `
      : `
    background: linear-gradient(#dae6f4, #d0def0);
    cursor: pointer;
    &:hover{
    opacity: 0.8
  }
  `}
`

export { TabPanel, Tab }
