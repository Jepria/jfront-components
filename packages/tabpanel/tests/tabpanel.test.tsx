import { render } from "@testing-library/react"
import React from "react"
import { Tab, TabPanel } from "../src"
import {
  Toolbar,
  ToolbarButtonCreate,
  ToolbarButtonDelete,
} from "@jfront/ui-toolbar"

test("Checking for the existence of an element TabPanel", () => {
  render(<TabPanel />)
})

test("Matches snapshot ", () => {
  const { asFragment } = render(
    <TabPanel>
      <Tab>Tab1</Tab>
      <Tab>Tab2</Tab>
    </TabPanel>,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("TabPanel with Toolbar renders correctly", () => {
  const tools = render(
    <>
      <TabPanel>
        <Tab onClick={() => {}}>Tab 1</Tab>
      </TabPanel>
      <Toolbar>
        <ToolbarButtonCreate />
        <ToolbarButtonDelete />
      </Toolbar>
    </>,
  )
  expect(tools.asFragment()).toMatchSnapshot()
})
