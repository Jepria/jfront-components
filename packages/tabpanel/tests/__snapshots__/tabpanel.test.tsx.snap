// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm gSRbKu"
  >
    <div
      class="sc-AxirZ ercANe"
    >
      Tab1
    </div>
    <div
      class="sc-AxirZ ercANe"
    >
      Tab2
    </div>
  </div>
</DocumentFragment>
`;

exports[`TabPanel with Toolbar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm gSRbKu"
  >
    <div
      class="sc-AxirZ ercANe"
    >
      Tab 1
    </div>
  </div>
  <nav
    class="sc-AxiKw kNbwFY"
  >
    <button
      class="sc-AxhCb MacuA"
    >
      <img
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAAXNSR0IArs4c6QAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9wBDAcfIC3l2C8AAAIRSURBVDjLlZLLThRRFEXXuffWq5s3RB5CogZRQ0hk4DsS/RaHOmHiNzjSqb/h0ImJ0eBQI1GDEBQSHtI00LR0dVVX3ePIEW2CZ7zPyk72Errc3cX7BHFAVAmWRS1Zms3l7ZylF+9OZV03gC88faNVHbs8QphF7PzY062NXemW7QoospKhmX4mHgwSblfI2wUba1ucGQCgTsjDDg1TI/UpRroWwPzjndBYem2VTqVNYXPEd4+6e4sL0BE0UzrSwePJmjk+U59TmJY0ybz3Ze65+fgOqqAo1lqsMzifFDDd0Z5rMeeYYqgcpshKdL5d/Ey/m4N6g2r/sL9wY0zjJMKZgNCG1HePONg/FNeW9Nmv6gadmQOujl8hMrP00cd2vu121mv4VSXT0lWux1DxOKNUbcTJR4ttmEUXa/KttZPRfAVfzq/wdWwFQtAW9Gz2M6CDMA3BhFCGIYXJ+e0atFfblK58JLee3ua450ibA3UkAkJF1HASpsXJZtMl8zAbzxdDO5MuiAwiBiPC8WaL1m4qLigiRurjMro3CQqqEOSO/bndcu3SJ4IpyD4Xpva6IT4p0L/zBQbjBPf++dtT0yw8eUie5MaNQHUiIVmvmDIWPrxcOqMHHrAgVYiiGGstqvo/IgGikEMepWhQAnJ2lb3zJLUKcTJI8qaXYk2QUM8OUKuEh5GMpxeXrTP41M5J0L3oHxh15JAxMjBeAAAAAElFTkSuQmCC"
      />
    </button>
    <button
      class="sc-AxhCb MacuA"
    >
      <img
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAAXNSR0IArs4c6QAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9wBDAcgO/9eO/8AAALLSURBVDjLpZFJb1MHFIW/+zzGTmLjxHYMqSIDCTFlaJO2EQgpQNXICCEQC6YFYscPoKv+iZIKiQ0rWlFVLUGt2HRRlaFFiujAUJEJoQRZcRwntuPk2X6237ss7AX7Xuks7uJcnXs+UVX+z4ghPhy1iAGT4cjVD8qF+VWH3wXIAH6gE/ACH/uNw088XQevbW7czAHeQBABN9DEB9QWc8rsv3yfTosXWG4fCAAGcPH2d6rbYgRPTUgVcHV0YEATgG9Gx5RcERJ7GD02rhbQaMsCPps4oRw4guzYxeTnaQWwbRsBsH56rN7FObh/DzsYwfXpJ8zcu8Xd5y/FD3wxMqYH0+eo//ccL00YSvEys8aBH2+IALjDfTxL9uuHc/PkGzXwBYj29lA8fYZaNktiepp8uQwVk6jXw1I4THJlXRSr1eS2RBKAN5EerYE+AH0IWhpMaT6R0Eegf4BWQGdBAbqCPeD3t17o6B+kmlkgCkx1dWppc4sVwNVG5QJ6ATFcXHFsWQV8oThWvYwsjR3lz+kHbAELgGcwRbpa0ZnMEvZ7vPdF4/zi9ohkMwzTipHojmCs2DZrQBEwgaObJbXMLYpAFagDJaBs1ZhwudRs7wVgDXDXHIdK23w5tl2t3DILCnuAaDt+FiiUN9htVrgUi+tvq7kWPRSjYTepgXs8GFQjn2VLYS8tBUZG8R86xH4gCRTsBp2FIuM9Ia1CoOkoRqNe56uvrze6TZOiKgNt89/hMGuvX00tzzz79mkkxEdAP5Bv1gmtb/Dl2fNmQx14cvICj3q79QXoa1AL9E4sqhmfe/7XUID74QCLHtc/d+J92gR9AzrXHdS/En06KYLhoKiIrAO7hlPcHR7meD7/1oYhB0FFsGFkvJCb/SEWJxnfTqls4oghiIGhCDaCDTI1tJOLE8ewlQFHFW3jchRsR1Nnhwb4ObUbB0RFcER4BzxKOdvITdVmAAAAAElFTkSuQmCC"
      />
    </button>
  </nav>
</DocumentFragment>
`;
